main/src/cgeo/geocaching/connector/gc/GCParser.java

```diff
@@ -1088,13 +1075,36 @@ public int compare(final PocketQuery left, final PocketQuery right) {
 
         while (matcherPocket.find()) {
             final String guid = Html.fromHtml(matcherPocket.group(1)).toString();
+            final String name = Html.fromHtml(matcherPocket.group(2)).toString();
+            int count = 0;
             try {
-                final Integer count = Integer.valueOf(matcherPocket.group(2));
-                downloadablePocketQueries.put(guid, count);
+                count = Integer.parseInt(matcherPocket.group(3));
             } catch (final NumberFormatException e) {
                 Log.w("GCParser.getDownloadablePocketQueries: cannot parse PQ cache count");
                 e.printStackTrace();
             }
+            final String lastGenerationString = matcherPocket.group(4);
+            Date lastGeneration = null;
+            try {
+                lastGeneration = GCLogin.parseGcCustomDate(lastGenerationString);
+            } catch (final ParseException e) {
+                Log.w("GCParser.getDownloadablePocketQueries: cannot parse PQ last generation date");
+                e.printStackTrace();
+            }
+
+            int daysRemaining = 0;
+            try {
+                final String daysRemainingString = matcherPocket.group(6);
+                if (daysRemainingString != null && !"last".equals(daysRemainingString)) {
+                    daysRemaining = Integer.parseInt(daysRemainingString);
+                }
+            } catch (final NumberFormatException e) {
+                Log.w("GCParser.getDownloadablePocketQueries: cannot parse PQ days remaining");
+                e.printStackTrace();
```

Comment:
Actually, Log.w(...) should have an overloaded version that takes the exception as additional argument.